trie/trieStorageManager.go

```diff
@@ -356,8 +355,8 @@ func (tsm *trieStorageManager) TakeSnapshot(
 		address:          address,
 		rootHash:         rootHash,
 		mainTrieRootHash: mainTrieRootHash,
-		errChan:          errChan,
-		leavesChan:       leavesChan,
+		errChan:          iteratorChannels.ErrChan,
```

Comment:
Pass the `iteratorChannels` here, instead of errChan and leavesChan

Reply:
:+1: